### ISSUE TITLE
GoInfo hangs on latest master

### ISSUE THREAD

**Issue by ldelossa** [fatih/vim-go#1914]
### What did you do? (required. The issue will be **closed** when not provided.)
Used GoInfo on a method 

### What did you expect to happen?
The method signature to be displayed

### What happened instead?
Just hangs with no response

### Configuration (**MUST** fill this out):

* vim-go version:
MASTER (issue is fixed when I checkout tag v1.18)

* `vimrc` you used to reproduce (use a *minimal* vimrc with other plugins disabled; do not link to a 2,000 line vimrc):

* Vim version (first three lines from `:version`):
NVIM v0.2.2

* Go version (`go version`):
❯ go version
go version go1.10.3 darwin/amd64

* Go environment (`go env`):
❯ go env
GOARCH="amd64"
GOBIN="/Users/louis/git/go/bin"
GOCACHE="/Users/louis/Library/Caches/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/louis/git/go"
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/mc/xwfdd41j1j74xfmcmphbwpgw0000gn/T/go-build418438257=/tmp/go-build -gno-record-gcc-switches -fno-common"

**Comment by bhcleek** [comment]
Does it work if after setting `g:go_info_mode` to `guru` ?

**Comment by ldelossa** [comment]
@bhcleek yup, and the syntax is now red along with showing me function details which I think is different also.

**Comment by bhcleek** [comment]
The output of `GoInfo` is different when using guru.

But you've confirmed that this is indeed a bug in `:GoInfo` when `g:go_info_mode` is `gocode`.

**Comment by ldelossa** [comment]
Cool I'll use guru for now and watch for updates.

**Comment by cboggs** [comment]
Hate to pop into a closed issue, but as of 0aa956f (master HEAD as of right now) this is still the behavior I'm seeing. I've updated binaries as well, stopped gocode, and dropped its cache just to be sure. It works in some cases (simple variables), but not functions. It also seems to be confused about what line numbers I've attempted to run `:GoInfo` on.

Happy to switch to guru for now, but wanted to mention that the commit to fix this issue doesn't appear to have done so. :-)
